pkg/webhook/mutation/pod_mutator/pod_mutator_test.go

```diff
@@ -236,13 +237,13 @@ func TestHandlePodReinvocation(t *testing.T) {
 
 		updated := podWebhook.handlePodReinvocation(context.Background(), mutationRequest)
 		require.False(t, updated)
-		failingMutator.(*dtwebhook.PodMutatorMock).AssertCalled(t, "Enabled", mutationRequest.BaseRequest)
-		failingMutator.(*dtwebhook.PodMutatorMock).AssertCalled(t, "Reinvoke", mutationRequest.ToReinvocationRequest())
+		failingMutator.(*mocks.PodMutator).AssertCalled(t, "Enabled", mutationRequest.BaseRequest)
```

Comment:
without `AssertExpectations()` (see [288](https://github.com/Dynatrace/dynatrace-operator/pull/2340/files#diff-130a5f5e79e744f15bb582d9ac4e4d97c3b8cd6722caeb0cc71054fb81e86ddbR288)) we don't know if `Injected` and `Mutate` functions declared by `createFailPodMutatorMock` are called / not called in this scenario.

We can add `Maybe()` to `On()` declarations if want to reuse `create*Mock` function in different scenarios. Then we have to clearly define our expectations using for example `AssertCalled`/`AssertNotCalled`.

Reply:
After having a look into AssertExpectations, what's your opinion on simply using
```go
		failingMutator.(*mocks.PodMutator).AssertNotCalled(t, "Injected", mock.Anything)
		failingMutator.(*mocks.PodMutator).AssertNotCalled(t, "Mutated", mock.Anything)
```
Because we use the same function in all tests to create the mock with `On()`, `Return()` and so on, hence `AssertExpectations()` can't pass because Injected and Mutated are not even called.

Reply:
Clarified in the call too :+1: